src/SlashCommands.tsx

```diff
@@ -62,9 +69,36 @@ export const CommandCategories = {
     "other": _td("Other"),
 };
 
+type RunFn = ((roomId: string, args: string, cmd: string) => {error: any} | {promise: Promise<any>});
+
 class Command {
-    constructor({name, args='', description, runFn, category=CommandCategories.other, hideCompletionAfterSpace=false}) {
-        this.command = '/' + name;
+    command: string;
+    aliases: string[];
+    args: undefined | string;
+    description: string;
+    runFn: undefined | RunFn;
+    category: string;
+    hideCompletionAfterSpace: boolean;
+
+    constructor({
+        command,
+        aliases=[],
+        args='',
+        description,
+        runFn=undefined,
+        category=CommandCategories.other,
+        hideCompletionAfterSpace=false,
+    }: {
+        command: string;
+        aliases?: string[];
+        args?: string;
+        description: string;
+        runFn?: RunFn;
+        category: string;
+        hideCompletionAfterSpace?: boolean;
+    }) {
+        this.command = command;
+        this.aliases = aliases;
         this.args = args;
         this.description = description;
         this.runFn = runFn;
```

Comment:
Much of this can be avoided with a constructor like:

```
constructor(
    private command: string,
    private aliases: string[],
    // etc...
) {}
```

Reply:
if not going this route, please consider breaking the constructor out to its own interface to at least remove the duplication:

```
interface CommandOpts {
    command: string;
    aliases?: string[];
    // etc...
}

constructor(opts: CommandOpts) {
    this.command = opts.command;
}
```

Reply:
(this applies to the modern diff, but github makes it difficult to move, sorry)

Reply:
That's cool but makes the number of optionals here a little messy

Sure, will go the interface route, only breaks WebStorm's default parameter value lookup :((